src/pages/ca-opening-balances/form/OpeningBalanceForm.js

```diff
@@ -0,0 +1,185 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect, useState } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { CashBankRepository } from 'src/repositories/CashBankRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+
+export default function OpeningBalanceForm({ labels, maxAccess, recordId, record }) {
+  const [editMode, setEditMode] = useState(!!recordId)
+
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: CashBankRepository.OpeningBalance.page
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: recordId || null,
+      fiscalYear: '',
+      accountId: '',
+      currencyId: '',
+      currencyRef: '',
+      amount: '',
+      ...record
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      fiscalYear: yup.string().required(' '),
+      accountId: yup.string().required(' '),
+      currencyId: yup.string().required(' '),
+      amount: yup.number().required(' ')
+    }),
+    onSubmit: async obj => {
+      const currencyId = formik.values.currencyId
```

Comment:
Can we use those 3 fields directly without initializing them before?

Reply:
the if statement should work after the value of them